UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.SubscriptionService/SubscriptionServiceJob.cs

```diff
@@ -56,7 +56,23 @@ public async Task ProcessQueueMessage([QueueTrigger("%SubscriptionStorageConfigu
                 }
                 catch (Exception e)
                 {
-                    subscriptionRequestResult.ProvisioningState = e.Message;
+                    subscriptionRequestResult.ProvisioningState = "Failed";
+
+                    externalNotificationEntity.ResponseStatusCode = _d365CallbackConfiguration.Value.FailedStatusCode;                    
+
+                    //other potential errors
+                    if (e.Message.Contains("Webhook validation handshake failed"))
+                    {
+                        int startIndex = e.Message.IndexOf("Webhook validation handshake failed");
+                        subscriptionRequestResult.ErrorMessage = e.Message.Substring(startIndex, e.Message.Length - startIndex);
+                        externalNotificationEntity.ResponseDetails = "Failed to add subscription @Time:" + Convert.ToString(DateTime.UtcNow) + " with exception " + subscriptionRequestResult.ErrorMessage;
+                    }
+                    else
```

Comment:
Can have comment here as "else  //other potential errors" for other potential errors

Reply:
Done. Thanks